snuba/datasets/discover.py

```diff
@@ -258,12 +263,13 @@ def __init__(self) -> None:
                     abstract_transactions_columns=self.__transactions_columns,
                 ),
             ),
-            abstract_column_set=(
-                self.__common_columns
-                + self.__events_columns
-                + self.__transactions_columns
-            ),
+            abstract_column_set=abstract_column_set,
             writable_storage=None,
+            # In terms of logical schema this is still a single table until we introduce
+            # entities.
+            column_resolver=SingleTableResolver(
+                abstract_column_set, ["tags_key", "tags_value"]
```

Comment:
Contrarily to bucketed columns, tags_key and tags_value, to me, should not be part of the logical schema. They are not column names, they are a special syntax for a function.
The solution to avoid this special case, to me, is to move the tags_key/tags_value resolution before column resolution. Though both options (this one and moving the tags_key resolution earlier) are ugly. None seems a lot better than the other.